delphi/app/graph.py

```diff
@@ -1,75 +1,280 @@
 
-class GraphEdge():
-    """
-    Contains information about each directed edge within the graph representation
-    """
+    def get_vertex(self, name):
+        """
+        Finds the vertex associated with the name passed in
+        :param name: The name associated with the vertex to find
+        :return: The vertex represented by name
+        """
+        for vertex in self.vertices:
+            if name == vertex.name:
+                return vertex
 
-    def __init__(self, source, dest, capacity=0):
+    def vertex_in_network(self, name):
+        """
+        Decides if a vertex is in the network based off of name
+        :param name: The name of the vertex to check for
+        :return: True if the vertex is already in the network; false otherwise
         """
-        Constructs the GraphEdge object.
+        for vertex in self.vertices:
+            if vertex.name == name:
+                return True
+        return False
 
-        Params:
-            source: source node
-            dest: destination node
-            capacity: default capacity of the edge
+    def get_edges(self):
+        """
+        Retrieves all the edges in the network
+        :return: The edges in the network
         """
+        all_edges = []
+        for vertex in self.network:
+            for edge_check in self.network[vertex]:
+                all_edges.append(edge_check)
+        return all_edges
 
-        self.source = source
-        self.dest = dest
-        self.capacity = capacity
-    
-    def update_capacity(self, capacity):
+    def add_vertex(self, name, source=False, sink=False):
+        """
+        Adds a vertex to the flow network
+        :param name: The identifier to be associated with the node
+        :param source: True if it's a source node, false otherwise
+        :param sink: True if it's a sink node, false otherwise
+        :return: Returns a string if error, nothing otherwise
         """
-        Updates the capacity of the edge. Cannot be greater than MAX_CAPACITY.
+        if source is True and sink is True:
+            return "Vertex cannot be source and sink"
+        if self.vertex_in_network(name):
+            return "ERROR"
+        if source:
+            if self.get_source() is not None:
+                return "ERROR"
+        if sink:
+            if self.get_sink() is not None:
+                return "ERROR"
+        new_vertex = Vertex(name, source, sink)
+        self.vertices.append(new_vertex)
+        self.network[new_vertex.name] = []
 
-        TODO: Have this do calculations on how to update the capacity rather than just setting it.
+    def add_edge(self, start, end, capacity):
         """
+        Adds a directed edge to the flow network
+        :param start: The node the edge is directed out of
+        :param end: The node the edge is directed into
+        :param capacity: The initial amount that can flow through the edge
+        :return: string if error occurs, nothing otherwise
```

Comment:
Returning a string or void is not easily consumable. Have it always return a boolean instead.